app/models/concerns/availability.rb

```diff
@@ -36,5 +21,99 @@ def active_routes
     def successful_routes
       client_opportunity_matches.success.map(&:match_route).uniq
     end
+
+    def available_as_candidate_for_any_route?
+      (
+        MatchRoutes::Base.available.pluck(:type) -
+        UnavailableAsCandidateFor.where(client_id: id).distinct.pluck(:match_route_type)
+      ).any?
+    end
+
+    def make_available_in(match_route:)
+      route_name = MatchRoutes::Base.route_name_from_route(match_route)
+      UnavailableAsCandidateFor.where(client_id: id, match_route_type: route_name).destroy_all
+    end
+
+    def make_available_in_all_routes
+      UnavailableAsCandidateFor.where(client_id: id).destroy_all
+    end
+
+    def make_unavailable_in(match_route:, user:, reason:, expires_at: default_unavailable_expiration_date, match: nil)
+      route_name = MatchRoutes::Base.route_name_from_route(match_route)
+
+      # Delete any previous unavailable_as_candidate_fors so we can track the reason for this one
+      unavailable_as_candidate_fors.where(match_route_type: route_name).destroy_all
+
+      unavailable_as_candidate_fors.where(match_route_type: route_name).create(
```

Comment:
should these raise on fail?
```suggestion
      # Delete any previous unavailable_as_candidate_fors so we can track the reason for this one
      unavailable_as_candidate_fors.where(match_route_type: route_name).each(&:desroy!)

      unavailable_as_candidate_fors.where(match_route_type: route_name).create!(
```